staging/src/k8s.io/mount-utils/mount_linux_test.go

```diff
@@ -545,3 +549,220 @@ func mountArgsContainOption(t *testing.T, mountArgs []string, option string) boo
 
 	return strings.Contains(mountArgs[optionsIndex], option)
 }
+
+func TestIsMountPointUsingOpenAt2(t *testing.T) {
+	tests := map[string]struct {
+		expectedErr        error
+		expectedMountPoint bool
+		// returns the path to be tested for.
+		// returns an error if something went wrong during prepare.
+		prepare     func(*testing.T) (string, error)
+		openat2Mock func(int, string, *unix.OpenHow) (fd int, err error)
+	}{
+		"path is not absolute": {
+			expectedErr:        errPathNotAbsolute,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				return "../hello/", nil
+			},
+		},
+		"path is a symlink": {
+			expectedErr:        unix.ELOOP,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				tmpDir, err := os.MkdirTemp("", "symlink")
+				if err != nil {
+					return "", err
+				}
+				t.Cleanup(func() {
+					os.RemoveAll(tmpDir)
+				})
+
+				f, err := os.CreateTemp(tmpDir, "some")
+				if err != nil {
+					return "", err
+				}
+
+				symlink := filepath.Join(tmpDir, "symlink")
+				err = os.Symlink(f.Name(), symlink)
+				if err != nil {
+					return "", err
+				}
+				return symlink, nil
+			},
+		},
+		"path is a mount point": {
```

Comment:
are you able to test a bind mount case? Since it requires root, manual test is fine.

Also a bind mount to a symlink etc?

Reply:
I added the respective tests here as well so that you can pull and test to run it manually if reqd.